lib/CppInterOp/Compatibility.h

```diff
@@ -413,6 +417,22 @@ template <typename T> inline T convertTo(clang::Value V) {
 }
 #endif // CPPINTEROP_USE_CLING
 
+inline void InstantiateClassTemplateSpecialization(
+    Interpreter& interp, clang::ClassTemplateSpecializationDecl* CTSD) {
+#ifdef CPPINTEROP_USE_CLING
+  cling::Interpreter::PushTransactionRAII RAII(&interp);
+#endif
+#if CLANG_VERSION_MAJOR < 20
+  interp.getSema().InstantiateClassTemplateSpecialization(
+      clang::SourceLocation::getFromRawEncoding(1), CTSD,
```

Comment:
Can we move that into an interface of some sort?

Reply:
Could you elaborate? How?
I have moved it into `Compatibility.h`.

Reply:
You can take a look how this was done for other cases and draw some inspiration from it?

Reply:
`InstantiateClassTemplateSpecialization` is not used anywhere else in the InterOp's codebase.

Reply:
Generally we move the `ifdef`ed code in `Compatibility.h` and we provide an interface to use outside.

Reply:
Yes. That is what I have done here.
The `ifdef` is in `Compatibility.h`, this is `Compatibility.h` file. And this provides the interface `compat::InstantiateClassTemplateSpecialization` that is used in `CppInterOp.cpp` at line 723. No `ifdef` in `CppInterOp.cpp`.

Reply:
Sorry I am blind..